__tests__/tests.ts

```diff
@@ -1329,6 +1330,99 @@ describe('htmlbars-inline-precompile', function () {
         const template = precompileTemplate('&times;');
       `);
     });
+
+    it('emits setComponentTemplate and templateOnlyComponent when polyfilling rfc931 in hbs format', function () {
+      plugins = [
+        [
+          HTMLBarsInlinePrecompile,
+          {
+            compiler,
+            targetFormat: 'hbs',
+            transforms: [color],
+          },
+        ],
+      ];
+
+      let transformed = transform(
+        `import { template } from '@ember/template-compiler'; 
+         import HelloWorld from 'somewhere';
+         export default template('<HelloWorld @color={{red}} />', { scope: () => ({ HelloWorld }) });`
+      );
+
+      expect(transformed).toEqualCode(`
+        import templateOnly from "@ember/component/template-only";
+        import { setComponentTemplate } from "@ember/component";
+        import { precompileTemplate } from "@ember/template-compilation";
+        import HelloWorld from "somewhere";
+        export default setComponentTemplate(precompileTemplate('<HelloWorld @color={{"#ff0000"}} />', { scope: () => ({ HelloWorld }), strictMode: true }), templateOnly());
+      `);
+    });
+
+    it('emits setComponentsTemplate when polyfilling rfc931 with hbs target', function () {
+      plugins = [
+        [
+          HTMLBarsInlinePrecompile,
+          {
+            compiler,
+            targetFormat: 'hbs',
+            transforms: [color],
+          },
+        ],
+      ];
+
+      let transformed = transform(
+        `
+         import { template } from '@ember/template-compiler'; 
+         import HelloWorld from 'somewhere';
+         export default class {
+           static {
+             template('<HelloWorld @color={{red}} />', { component: this, scope: () => ({ HelloWorld }) });
```

Comment:
Looking at the RFC to reconcile this implementation and it got me wondering, is there value in a test that shows using the `instance` arg that's passed in to the `scope` function, say, for accessing private fields?

This is the code that got me wondering this:

```js
// in class member position, we can also put private fields in scope
class extends Component {
  static {
    template(
      "<Section @title={{this.title}} @subhead={{this.#secret}} />", 
      {
        component: this,
        scope: (instance) => ({ 
          Section, 
          "#secret": instance.#secret
        }),
      },
    )
  }
}
```